src/lib.rs

```diff
@@ -285,6 +288,25 @@ impl WalkDir {
         self.opts.max_open = n;
         self
     }
+
+    /// Set a compare function.
+    /// If a compare function is set, WalkDir will return all paths in sorted
+    /// order.
+    ///
+    /// ```rust,no-run
+    /// use std::cmp;
+    /// use std::ffi::OsString;
+    /// use walkdir::WalkDir;
+    ///
+    /// fn string_compare(a: &OsString, b: &OsString) -> cmp::Ordering {
+    ///     a.cmp(b)
+    /// }
+    /// WalkDir::new("foo").sort(Some(string_compare));
+    /// ```
+    pub fn sort(mut self, sorter: Option<fn(&OsString,&OsString) -> std::cmp::Ordering>) -> Self {
```

Comment:
I think the convention here is to name this `sort_by`. Could you rename it?

Also, the type signature doesn't look right to me and it will prevent using a closure. I'd expect something like this instead:

```rust
pub fn sort_by<F: FnMut(&OsString, &OsString) -> Ordering>(mut self, cmp: F) -> Self
```

This will probably require boxing `cmp` before storing it in `self.opts.sorter`. This is also going to require introducing a type parameter into the `WalkDir` type. Ug. Hmmm.